lib/Controller/SettingsController.php

```diff
@@ -142,14 +132,40 @@ public function update($wopi_url, $edit_groups, $doc_format, $test_wopi_url, $te
 			$this->appConfig->setAppValue('watermark_text', $watermark_text);
 		}
 
+		if ($zotero !== null) {
+			$this->appConfig->setAppValue('zotero', $zotero);
+		}
+
 		$richMemCache = \OC::$server->getMemCacheFactory()->create('richdocuments');
 		$richMemCache->clear('discovery.xml');
 
-		$response = [
+		return new DataResponse([
 			'status' => 'success',
 			'data' => ['message' => (string) $message]
-		];
+		]);
+	}
 
-		return $response;
+	/**
+	 * @NoAdminRequired
+	 * @UseSession
+	 *
+	 * @param string $zoteroAPIPrivateKey
+	 * @return DataResponse
+	 */
+	public function setPersonalSettings($zoteroAPIPrivateKey) {
+		$message = $this->l10n->t('Saved');
+		
+		$uid = $this->userSession->getUser()->getUID();
+
+		if ($zoteroAPIPrivateKey !== null) {
+			$this->appConfig->setUserValue($uid, 'zoteroAPIPrivateKey', $zoteroAPIPrivateKey);
+		}
+
+		return new DataResponse(
+			[
+				'status' => 'success',
+				'data' => ['message' => (string) $message]
```

Comment:
Unless we expect specific information from the server, I think translations are better handled at the client's side. Just for a "saved" message, the client can check the status of the response and provide a translated message by itself.

Reply:
this is something I would prefer not to touch.

Reply:
Maybe include a TODO comment?

Reply:
good point, done